packages/block-library/src/list-item/edit.native.js

```diff
@@ -85,21 +85,32 @@ export default function ListItemEdit( {
 	const blockProps = useBlockProps( {
 		...( hasInnerBlocks && styles[ 'wp-block-list-item__nested-blocks' ] ),
 	} );
+
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		allowedBlocks: [ 'core/list' ],
 		renderAppender: false,
 	} );
 
-	const placeholderWithPreferredScheme = usePreferredColorSchemeStyle(
+	// Set default placeholder text color from light/dark scheme or base colors
+	const defaultPlaceholderFromScheme = usePreferredColorSchemeStyle(
 		styles[ 'wp-block-list-item__list-item-placeholder' ],
 		styles[ 'wp-block-list-item__list-item-placeholder--dark' ]
 	);
 
-	// Add 0.62 hex opacity to baseColor text (e.g. #FFFFFF9E)
-	// or fallback to default color from stylesheet ($gray, $gray-50)
 	const defaultPlaceholderTextColor = style?.baseColors?.color?.text
-		? style.baseColors.color.text + '9E'
-		: placeholderWithPreferredScheme?.color;
+		? styles.color || style.baseColors.color.text
```

Comment:
I think we should have this condition separately, because for standard themes you won't have `style.baseColors.color.text` so it will pick the default color when you could still set a custom color from the color settings.

So maybe we could do something like:

```javascript
	const currentTextColor = style?.color || style?.baseColors?.color?.text;
```

And then use `currentTextColor` in `defaultPlaceholderTextColor`. What do you think?